pandas/core/arrays/categorical.py

```diff
@@ -618,16 +618,37 @@ def from_codes(cls, codes, categories, ordered=False):
         ----------
         codes : array-like, integers
             An integer array, where each integer points to a category in
-            categories or -1 for NaN
-        categories : index-like
+            categories or dtype.categories, or else is -1 for NaN
+        categories : index-like, optional
             The categories for the categorical. Items need to be unique.
-        ordered : boolean, (default False)
-            Whether or not this categorical is treated as a ordered
+            .. versionchanged:: 0.24.0
+
+                The `categories` parameter has been made optional.
+        ordered : bool, optional
+            Whether or not this categorical is treated as an ordered
             categorical. If not given, the resulting categorical will be
             unordered.
+
+            .. versionchanged:: 0.24.0
+
+                The default value has been changed to  ``None``. Previously
```

Comment:
I don't think this is necessary. The default value of "None" ends up being False, right? So there's no real change in behavior.